usb/esp_tinyusb/descriptors_control.c

```diff
@@ -124,27 +125,15 @@ void tusb_set_descriptor(const tusb_desc_device_t *dev_desc, const char **str_de
              dev_desc->idVendor, dev_desc->idProduct, dev_desc->bcdDevice,
              dev_desc->iManufacturer, dev_desc->iProduct, dev_desc->iSerialNumber,
              dev_desc->bNumConfigurations);
-    s_device_descriptor = *dev_desc;
-    s_configuration_descriptor = cfg_desc;
-
-    if (str_desc != NULL) {
-        memcpy(s_str_descriptor, str_desc,
-               sizeof(s_str_descriptor[0])*USB_STRING_DESCRIPTOR_ARRAY_SIZE);
-    }
-}
 
-tusb_desc_device_t *tusb_get_active_desc(void)
-{
-    return &s_device_descriptor;
-}
-
-char **tusb_get_active_str_desc(void)
-{
-    return s_str_descriptor;
+    // Save passed descriptors
+    s_device_descriptor = dev_desc;
+    s_configuration_descriptor = cfg_desc;
+    memcpy(s_str_descriptor, str_desc, str_desc_count * sizeof(str_desc[0]));
 }
 
-void tusb_clear_descriptor(void)
+void tinyusb_set_str_descriptor(const char *str, int str_idx)
 {
-    memset(&s_device_descriptor, 0, sizeof(s_device_descriptor));
-    memset(&s_str_descriptor, 0, sizeof(s_str_descriptor));
+    assert(str_idx < USB_STRING_DESCRIPTOR_ARRAY_SIZE);
```

Comment:
According to this code we can add a special string descriptors within the following numbers = [0 .. 7]. 
Based on the theory that any string descriptor can be present in the device configuration (for example 0xEE for Microsoft OS Descriptors support) probably it is better to make the list of the string descriptors with the number, length and string data. 

But if the NCM does not need special string number and mac addr can be saved in slots under 5,6,7 numbers...that can work then.

Reply:
Good point, I didn't take these special string descriptors into account.

I haven't encountered too many uses of Microsoft Descriptors, so I'm tempted to postpone this breaking feature...

Do you know some example usages?

Reply:
Honestly, no.
So it is nice to have option to set 0xEE string descriptors but I have never seen them in real life.
But not neccesary I suppose.

Reply:
OK, let's leave it to another PR :)